src/test/libcephfs/test.cc

```diff
@@ -2469,3 +2470,74 @@ TEST(LibCephFS, Lseek) {
   ASSERT_EQ(0, ceph_close(cmount, fd));
   ceph_shutdown(cmount);
 }
+
+TEST(LibCephFS, LookupVino) {
+  struct ceph_mount_info *cmount;
+  ASSERT_EQ(ceph_create(&cmount, NULL), 0);
+  ASSERT_EQ(ceph_conf_read_file(cmount, NULL), 0);
+  ASSERT_EQ(0, ceph_conf_parse_env(cmount, NULL));
+  ASSERT_EQ(ceph_mount(cmount, NULL), 0);
+
+  char dir_path[64];
+  char snap_name[64];
+  char snapdir_path[128];
+  char snap_path[256];
+  char file_path[PATH_MAX];
+  char snap_file[PATH_MAX];
+  sprintf(dir_path, "/dir0_%d", getpid());
+  sprintf(snap_name, "snap0_%d", getpid());
+  sprintf(file_path, "%s/file_%d", dir_path, getpid());
+  sprintf(snapdir_path, "%s/.snap", dir_path);
+  sprintf(snap_path, "%s/%s", snapdir_path, snap_name);
+  sprintf(snap_file, "%s/file_%d", snap_path, getpid());
+
+  ASSERT_EQ(0, ceph_mkdir(cmount, dir_path, 0755));
+  int fd = ceph_open(cmount, file_path, O_WRONLY|O_CREAT, 0666);
+  ASSERT_LE(0, fd);
+  ASSERT_EQ(0, ceph_close(cmount, fd));
+  ASSERT_EQ(0, ceph_mksnap(cmount, dir_path, snap_name, 0755, nullptr, 0));
```

Comment:
@jtlayton the build is failing here, as mentioned in,
https://github.com/ceph/ceph/pull/40769#issuecomment-846441305

`ceph_mksnap` was introduced in PR https://github.com/ceph/ceph/pull/37721/  The PR doesn't seem simple to backport.  Can the the test be modified to just do a mkdir of a '.snap'  directory explicitly and avoid the need of 'ceph_mksnap`?

Reply:
>  just do a mkdir of a '.snap' directory explicitly

@ajarr Yes, that's what @jtlayton suggested in the discussion we had in the octopus backport (which is identical to this one).

All that is needed is the actual code to replace the `ceph_mksnap` call.

Reply:
Ok, just added a proposed patch. Let's see if it builds...

Reply:
@jtlayton @smithfarm thanks!